tesseract/tesseract_scene_graph/include/tesseract_scene_graph/srdf_model.h

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file srdf_model.h
+ * @brief Parse srdf xml
+ *
+ * @author Levi Armstrong, Ioan Sucan
+ * @date May 12, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TESSERACT_SCENE_GRAPH_SRDF_MODEL_H
+#define TESSERACT_SCENE_GRAPH_SRDF_MODEL_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <utility>
+#include <memory>
+#include <tinyxml2.h>
+#include <console_bridge/console.h>
+#include <tesseract_scene_graph/graph.h>
+#include <fstream>
+#include <array>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+/// Main namespace
+namespace tesseract_scene_graph
+{
+/** @brief A structure to hold opw kinematics data */
+struct OPWKinematicParameters
+{
+  double a1{ 0 }, a2{ 0 }, b{ 0 }, c1{ 0 }, c2{ 0 }, c3{ 0 }, c4{ 0 };
+  double offsets[6]{ 0, 0, 0, 0, 0, 0 };
+  signed char sign_corrections[6]{ 1, 1, 1, 1, 1, 1 };
+};
+
+/** @brief Representation of semantic information about the robot */
+class SRDFModel
+{
+public:
+  using Ptr = std::shared_ptr<SRDFModel>;
+  using ConstPtr = std::shared_ptr<const SRDFModel>;
+
+  using JointState = std::unordered_map<std::string, double>;
+  using JointStates = std::unordered_map<std::string, JointState>;
+  using GroupStates = std::unordered_map<std::string, JointStates>;
+  using TCPs = std::unordered_map<std::string, Eigen::Isometry3d>;
+  using GroupTCPs = std::unordered_map<std::string, TCPs>;
+  using ChainGroups = std::unordered_map<std::string, std::vector<std::pair<std::string, std::string>>>;
+  using JointGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using LinkGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using GroupOPWKinematics = std::unordered_map<std::string, OPWKinematicParameters>;
```

Comment:
I suppose this is fairly common, but would it be better to have a group that is an arbitrary set of of DH parameters? We could still use it with OPWKinematics if it was an OPW robot.

Reply:
I guess what I am saying is what do you think about replacing OPWKinematicParameters with a DH Table?